### PR TITLE
Sdk update

### PR DESCRIPTION
**Pull request by robert-zaremba**
* update SDK to the lasts version
* Use constants for error messages
* added few helper functions
* updating tests to adopt breaking changes in `near_sdk_sim` (work in progress)

### CODE REVIEW

File: ref-exchange/src/account_deposit.rs [hunk 1]
```diff
@@ -26,25 +26,23 @@ pub struct AccountDeposit {
 impl AccountDeposit {
     /// Adds amount to the balance of given token while checking that storage is covered.
     pub fn add(&mut self, token: AccountId, amount: Balance) {
-        let prev_amount = *self.tokens.get(&token).unwrap_or(&0);
-        self.tokens.insert(token, prev_amount + amount);
-        assert!(
-            self.storage_usage() <= self.amount,
-            "ERR_INSUFFICIENT_STORAGE"
-        );
+        if let Some(x) = self.tokens.get_mut(&token) {
```
Comment (robert-zaremba) [inline ref-exchange/src/account_deposit.rs:29]: using `get_mut` is an optimization - in most of the cases we don't need the tokens will be already registered.

File: ref-exchange/src/errors.rs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+pub const ERR10_ACC_NOT_REGISTERED: &str = "E20: account not registered";
+pub const ERR11_INSUFFICIENT_STORAGE: &str = "E10: insufficient $NEAR storage deposit";
+pub const ERR12_TOKEN_NOT_WHITELISTED: &str = "E12: token not whitelisted";
+
+// Account Deposits //
+
+pub const ERR21_TOKEN_NOT_REG: &str = "E21: token not registered";
+pub const ERR22_NOT_ENOUGH_TOKENS: &str = "E23: not enough tokens in deposit";
+// pub const ERR23_NOT_ENOUGH_NEAR: &str = "E23: not enough NEAR in deposit";
+pub const ERR24_NON_ZERO_TOKEN_BALANCE: &str = "E24: non-zero token balance";
```
Comment (robert-zaremba) [inline ref-exchange/src/errors.rs:12]: For the moment I defined errors only related to storage and account deposits.
If this makes sense I will update other errors as well in other PR.

File: ref-exchange/src/errors.rs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// Storage errors //
+
+pub const ERR10_ACC_NOT_REGISTERED: &str = "E20: account not registered";
```
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply ref-exchange/src/errors.rs:3]: The E20 - ERR10 mismach is a mistake. (copy paste). The idea of having numbers is to group them by error type:
* 1x account registration and storage
* 2x deposits
* ....

If you don't like numbers in errors then I can remove it. LMK.

I don't think it makes sense to use enum. Error messages are strings, and used as strings, not enums. Especially, that for some of them we want to have formatting (eg: errors with `_F` suffix).

File: ref-exchange/src/owner.rs [hunk 4]
```diff
@@ -74,4 +68,12 @@ impl Contract {
             whitelisted_tokens: UnorderedSet::new(b"w".to_vec()),
         }
     }
+
+    pub(crate) fn only_owner(&self) {
```
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply ref-exchange/src/owner.rs:72]: OK, `only_owner` is a pattern in Ethereum. I will rename.

File: ref-exchange/src/account_deposit.rs [hunk 5]
```diff
@@ -7,13 +7,24 @@ use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
...
 
-use crate::utils::{ext_fungible_token, ERR_NOT_REGISTERED, GAS_FOR_FT_TRANSFER};
+use crate::utils::{ext_fungible_token, GAS_FOR_FT_TRANSFER};
 use crate::*;
 
 const MAX_ACCOUNT_LENGTH: u128 = 64;
 const MIN_ACCOUNT_DEPOSIT_LENGTH: u128 = MAX_ACCOUNT_LENGTH + 16 + 4;
 
 /// Account deposits information and storage cost.
+#[cfg(not(test))]
```
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply ref-exchange/src/account_deposit.rs:17]: Thanks for tip!

[the remark this answers is not in the document]
Comment (robert-zaremba) [reply ref-exchange/src/account_deposit.rs]: done

File: ref-exchange/build.sh [hunk 6]
```diff
@@ -2,4 +2,5 @@
 set -e
 
 RUSTFLAGS='-C link-arg=-s' cargo +stable build --target wasm32-unknown-unknown --release
-cp target/wasm32-unknown-unknown/release/ref_exchange.wasm ../res/
+cd ..
+cp target/wasm32-unknown-unknown/release/ref_exchange.wasm ./res/
```
Comment (ilblackdragon) [inline ref-exchange/build.sh:6]: why not common res folder outside?

File: ref-exchange/src/errors.rs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+// Storage errors //
```
Comment (ilblackdragon) [inline ref-exchange/src/errors.rs:1]: let's not use // at the end
/// is for doc strings
//! for file doc strings

File: ref-exchange/src/errors.rs [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+pub const ERR10_ACC_NOT_REGISTERED: &str = "E10: account not registered";
+pub const ERR11_INSUFFICIENT_STORAGE: &str = "E11: insufficient $NEAR storage deposit";
+pub const ERR12_TOKEN_NOT_WHITELISTED: &str = "E12: token not whitelisted";
+
+// Account Deposits //
+
+pub const ERR21_TOKEN_NOT_REG: &str = "E21: token not registered";
+pub const ERR22_NOT_ENOUGH_TOKENS: &str = "E23: not enough tokens in deposit";
+// pub const ERR23_NOT_ENOUGH_NEAR: &str = "E23: not enough NEAR in deposit";
```
Comment (ilblackdragon) [inline ref-exchange/src/errors.rs:11]: let's avoid checking in commented code to keep code clean

File: ref-exchange/build.sh (hunk 6 above)
Comment (robert-zaremba) [reply ref-exchange/build.sh:6]: It's is a common directory. When using workspace, the current directory is the workspace directory.

File: ref-exchange/src/errors.rs (hunk 7 above)
Comment (robert-zaremba) [reply ref-exchange/src/errors.rs:1]: OK. BTW: this is not a doc string.

File: ref-exchange/src/errors.rs (hunk 8 above)
Comment (robert-zaremba) [reply ref-exchange/src/errors.rs:11]: I will use it in the next PR.

**Pull request merged**